### PR TITLE
[TACHYON-808] Use Precondition checks for TachyonFS#validateUri

### PR DESCRIPTION
**Pull request by miasantreble**
https://tachyon.atlassian.net/browse/TACHYON-808

### CODE REVIEW

File: integration-tests/src/test/java/tachyon/client/TachyonFSIntegrationTest.java [hunk 1]
```diff
@@ -95,7 +95,7 @@ public void createFileWithFileAlreadyExistExceptionTest() throws IOException {
     fileId = sTfs.createFile(uri);
   }
 
-  @Test(expected = IOException.class)
+  @Test(expected = IllegalArgumentException.class)
```
Comment (apc999) [inline integration-tests/src/test/java/tachyon/client/TachyonFSIntegrationTest.java]: Actually this is still JUNIT3 way. Can you make it in JUNIT4 way, by using 
```java
@Rule
public ExpectedException mThrown= ExpectedException.none();
```
and in test case, do:
```java
mThrown.expect(IllegalArgumentException.class);
mThrown.expectMessage("what ever the message is.");
```

Comment (miasantreble) [inline integration-tests/src/test/java/tachyon/client/TachyonFSIntegrationTest.java]: done

File: clients/unshaded/src/main/java/tachyon/client/TachyonFS.java [hunk 2]
```diff
@@ -981,27 +981,21 @@ public synchronized void updateRawTableMetadata(int id, ByteBuffer metadata) thr
   }
 
   /**
-   * Validates the given uri, throwing an IOException if the uri is invalid.
+   * Validates the given uri, throws an appropriate Exception if the uri is invalid.
    *
    * @param uri The uri to validate
    */
-  private void validateUri(TachyonURI uri) throws IOException {
+  private void validateUri(TachyonURI uri) {
```
Comment (apc999) [inline clients/unshaded/src/main/java/tachyon/client/TachyonFS.java]: the following definition of `err` is no more needed.

Comment (miasantreble) [inline clients/unshaded/src/main/java/tachyon/client/TachyonFS.java]: done

**Comment by apc999** [comment]
+1 LGTM

**Comment by calvinjia** [comment]
+1 LGTM

**Pull request merged**